packages/docs/docs/lambda/aws-sqs-lambda-integration.md

```diff
@@ -0,0 +1,368 @@
+
+```js title="create queue"
+// 👇 create the queue
+const remotionQueue = new sqs.Queue(this, "queue", {
+  encryption: sqs.QueueEncryption.KMS_MANAGED,
+  queueName: "remotion_queue",
+});
+```
+
+- `remotion_queue` grants access to 2 Lambda functions to interact with it. Each individual role are assigned to their respective Lambda function.
+
+  ```js title="apiIntegrationRole"
+  // 👇 create the apiIntegrationRole role
+  const apiIntegrationRole = new IAM.Role(this, "api-integration-role", {
+    assumedBy: new IAM.ServicePrincipal("lambda.amazonaws.com"),
+    managedPolicies: [
+      ManagedPolicy.fromAwsManagedPolicyName(
+        "service-role/AWSLambdaBasicExecutionRole"
+      ),
+    ],
+  });
+  ```
+
+- This role is assigned to [`enqueue-function`](https://github.com/alexfernandez803/remotion-serverless/blob/main/apigw-sqs-app/lib/remotion-cdk-starter-stack.ts#L86), to allow it to write CloudWatch logs.
+
+  ```js title="remotionSQSLambdaRole"
+  // 👇 create a role with custom name
+  const renderFunctionLambdaRole = new Role(this, "remotionSQSLambdaRole", {
+    roleName: "remotionSQSLambdaRole",
+    assumedBy: new ServicePrincipal("lambda.amazonaws.com"),
+    managedPolicies: [
+      ManagedPolicy.fromAwsManagedPolicyName(
+        "service-role/AWSLambdaBasicExecutionRole"
+      ),
+      ManagedPolicy.fromManagedPolicyName(
+        this,
+        "remotion-executionrole-policy",
+        "remotion-executionrole-policy"
+      ),
+    ],
+  });
+  ```
+
+- The [`render-lambda-function`](https://github.com/alexfernandez803/remotion-serverless/blob/main/apigw-sqs-app/lib/remotion-cdk-starter-stack.ts#L97) has been assigned this role, which includes access to interact with Cloudwatch and permission to access other AWS services as specified in the `remotion-executionrole-policy`.
+
+  ```js title="Grant access to the queue"
+  // 👇 Grant permission to publish to the queue
+  remotionQueue.grantSendMessages(apiIntegrationRole);
+  // 👇 grant permission to consume messages from the queue
+  remotionQueue.grantConsumeMessages(renderFunctionLambdaRole);
+  ```
+
+- Allow the render function to listen to the queue.
+
+  ```js title="Listen to queue"
+  remotionRenderFunction.addEventSource(
+    new SqsEventSource(remotionQueue, {
+      batchSize: 1,
+      maxBatchingWindow: Duration.minutes(5),
+      reportBatchItemFailures: true, // default to false
+    })
+  );
+  ```
+
+### Interacting with the API
+
+The API requires an authorization token to interact with it. To obtain the token, first go to the Serverless dashboard to retrieve outputs such as `apigw-sqs-app-stack.region`, `apigw-sqs-app-stack.userPoolClientId`, and `apigw-sqs-app-stack.userPoolId`, which are used to authenticate with Cognito. If you do not have a frontend application, you can create a user and an authentication token manually for the API by following this [guide](docs/lambda/without-iam/example#test-your-endpoint).
```

Comment:
Does this require the Serverless framework as well? Can we make it not require that?

Reply:
I've pushed an update on this.